README.md

```diff
@@ -82,74 +88,66 @@ Drop-In is currently localized for [23 languages](https://github.com/braintree/b
+dropInRequest.threeDSecureRequest = threeDSecureRequest
 ```
 
 ### Fetch last used payment method
 
-If your user already has an existing payment method, you may not need to show Drop-In. You can check if they have an existing payment method using `DropInResult#fetchDropInResult`. Note that a payment method will only be returned when using a client token created with a `customer_id`.
+If your user already has an existing payment method, you may not need to show Drop-in. You can check if they have an existing payment method using `DropInClient#fetchMostRecentPaymentMethod`. A payment method will only be returned when using a client token created with a `customer_id`.
 
-```java
-DropInResult.fetchDropInResult(activity, clientToken, new DropInResult.DropInResultListener() {
-    @Override
-    public void onError(Exception exception) {
-        // an error occurred
-    }
-
-    @Override
-    public void onResult(DropInResult result) {
-        if (result.getPaymentMethodType() != null) {
+```kotlin
+val dropInClient = DropInClient(this, "<#CLIENT_TOKEN_WITH_CUSTOMER_ID>", dropInRequest)
+    dropInClient.fetchMostRecentPaymentMethod(this) { dropInResult, error ->
+        error?.let {
+            // an error occurred
+        }
+        dropInResult?.let { result ->
+            result.paymentMethodType?.let { paymentMethodType ->
             // use the icon and name to show in your UI
-            int icon = result.getPaymentMethodType().getDrawable();
-            int name = result.getPaymentMethodType().getLocalizedName();
-
-            if (result.getPaymentMethodType() == PaymentMethodType.GOOGLE_PAY) {
-                // The last payment method the user used was GooglePayment. The GooglePayment
-                // flow will need to be performed by the user again at the time of checkout
-                // using GooglePayment#requestPayment(...). No PaymentMethodNonce will be
-                // present in result.getPaymentMethodNonce(), this is only an indication that
-                // the user last used GooglePayment.
+            val icon = paymentMethodType.drawable
+            val name = paymentMethodType.localizedName
+                
+            if (paymentMethodType == DropInPaymentMethod.GOOGLE_PAY) {
+                // The last payment method the user used was Google Pay.
+                // The Google Pay flow will need to be performed by the
+                // user again at the time of checkout.
             } else {
-                // show the payment method in your UI and charge the user at the
-                // time of checkout using the nonce: paymentMethod.getNonce()
-                PaymentMethodNonce paymentMethod = result.getPaymentMethodNonce();
+                // Use the payment method show in your UI and charge the user
+                // at the time of checkout.
+                val paymentMethod = result.paymentMethodNonce
             }
-        } else {
+        } ?: run {
```

Comment:
I think `let?. {} ?: run {}` is the correct Kotlin syntax for this block - updated the docs PR for this too, correct me if I'm wrong.

Reply:
Yeah that would be equivalent to an `if/let/else` similar to Swift.

This may be a code smell though. For some reason, GooglePay nonces aren't vaulted so we return payment method type as a workaround. For other payment methods, we return a nonce (without payment method type). There could be a bug internally with the `DropInResult` we callback.

Reply:
Ah nm after discussion we are good I missed the part [here](https://github.com/braintree/braintree-android-drop-in/blob/f9ad3aca587509a1138c73afbdd80d8bd7970dbd/Drop-In/src/main/java/com/braintreepayments/api/DropInResult.java#L47) where we're setting `paymentMethodType` internally on `DropInResult` after setting the nonce.